src/graph/expression_operators.h

```diff
@@ -3,75 +3,346 @@
+ */
+/**
+* @brief Sine
+*
+* Computes the element-wise sine of the expression: \f$ \sin(a) \f$
+* @see SinNodeOp
+*/
 Expr sin(Expr a);
-Expr cos(Expr a);
-Expr tan(Expr a);
 
-Expr clip(Expr a, float c);
+/**
+* @brief Cosine
+*
+* Computes the element-wise cosine of the expression: \f$ \cos(a) \f$
+* @see CosNodeOp
+*/
+Expr cos(Expr a);
 
+/**
+* @brief Tangent
+*
+* Computes the element-wise tangent of the expression: \f$ \tan(a) \f$
+* @see TanNodeOp
+*/
+Expr tan(Expr a);
+/** @} */
+/** @} */
+
+
+/**
+ * @addtogroup graph_ops_arithmetic Arithmetic
+ * @ingroup graph_ops
+ * @brief Performs arithmetic in the expression graph
+ *
+ * @{
+*/
+
+/**
+ * Returns \f$ -a \f$
+ * @see NegNodeOp for implementation.
+ */
+///@{
 Expr operator-(Expr a);
+///@}
 
 /*********************************************************/
 
-Expr operator+(Expr a, Expr b);
-Expr operator+(float a, Expr b);
-Expr operator+(Expr a, float b);
-
-Expr operator-(Expr a, Expr b);
-Expr operator-(float a, Expr b);
-Expr operator-(Expr a, float b);
-
-Expr operator*(Expr a, Expr b);
-Expr operator*(float a, Expr b);
-Expr operator*(Expr a, float b);
+/**
+ * @name Addition
+ * @brief Performs \f$ a + b \f$ in the expression graph
+ *
+ * @{
+*/
+Expr operator+(Expr a, Expr b);   //!< @see Implementation in PlusNodeOp
+Expr operator+(float a, Expr b);  //!< @see Implementation in ScalarAddNodeOp
+Expr operator+(Expr a, float b);  //!< @see Implementation in ScalarAddNodeOp
+/** @} */
+
+/**
+ * @name Subtraction
+ * @brief Performs \f$ a - b \f$ in the expression graph
+ *
+ * @{
+*/
+Expr operator-(Expr a, Expr b);   //!< @see Implementation in MinusNodeOp
+Expr operator-(float a, Expr b);  //!< @see Implementation in ScalarAddNodeOp
+Expr operator-(Expr a, float b);  //!< @see Implementation in ScalarAddNodeOp
+/** @} */
+
+/**
+ * @name Multiplication
+ * @brief Performs \f$ a * b \f$ in the expression graph
+ *
+ * @{
+*/
+Expr operator*(Expr a, Expr b);   //!< @see Implementation in MultNodeOp
+Expr operator*(float a, Expr b);  //!< @see Implementation in ScalarMultNodeOp
+Expr operator*(Expr a, float b);  //!< @see Implementation in ScalarMultNodeOp
+/** @} */
+
+/**
+ * @name Division
+ * @brief Performs \f$ a / b \f$ in the expression graph
+ *
+ * @{
+*/
+Expr operator/(Expr a, Expr b);   //!< @see Implementation in DivNodeOp
+Expr operator/(float a, Expr b);  //!< Promotes @p a to Expression<ConstantNode> and uses operator/(Expr a, Expr b).
+                                  //!< @todo efficient version of this without ExpressionGraph::constant
+Expr operator/(Expr a, float b);  //!< Implementation via \f$ a * \frac{1}{b} \f$.
+/** @} */
+
+/** @} */
+
+/**
+ * @ingroup graph_ops_mathematical
+ * @brief Computes the square root of an expression.
+ *
+ * Evaluates \f$\sqrt{a + \mathrm{eps}} \f$ element-wise on the expression
+ *
+ * @param a Expression to square root
+ * @param eps Optional positive epsilon to avoid domain errors for small values in @p a
+ */
```

Comment:
In #788 we use automatic `@brief`s, can you check if it works with `@ingroup`? Does `@ingroup` work as the very first line only or can we use it later too?

Reply:
In group can be used later. I've update this.